monai/bundle/scripts.py

```diff
@@ -116,6 +121,212 @@ def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int
+                extractall(filepath=filepath, output_dir=bundle_dir_, has_base=True)
+        else:
+            download_url(url=url_, filepath=bundle_dir_, hash_val=None, progress=progress_)
+            if extract_ is True:
+                logger.info("When download a whole bundle package, extract is not supported, skip extracting.")
+    elif source_ == "github":
+        _download_from_github(
+            repo=repo_,
+            tag_name=package_,
+            download_path=bundle_dir_,
+            filename=filename_,
+            progress=progress_,
+            extract=extract_,
+        )
+    else:
+        raise NotImplementedError("So far, only support to download from url or `github` source.")
+
+
+def load(
+    weights_name: str = "model.pt",
+    is_ts_model: bool = False,
+    package: Optional[str] = None,
+    bundle_dir: PathLike = ".",
+    source: str = "github",
+    repo: Optional[str] = None,
+    url: Optional[str] = None,
+    progress: bool = True,
+    extract: bool = False,
+    map_location=None,
+):
+    """
+    Download (if necessary) and load model weights.
+
+    Args:
+        weights_name: the name of the weights file that will be loaded.
+        is_ts_model, a flag to specify if the weights file is a TorchScript module.
+        package: the bundle package name.
+            If the weights need to be downloaded first and `url` is `None`, it must be provided.
+        bundle_dir: the directory the weights will be loaded from.
+        source: the place that saved the bundle package.
+            If `source` is `github`, the bundle package should be within the releases.
+        repo: the repo name. If the weights need to be downloaded first and `url` is `None`, it must be provided.
+        url: the url to download the data.
+        progress: whether to display a progress bar when downloading.
+        extract: whether to extract the downloaded file.
+        map_location: pytorch API parameter for `torch.load` or `torch.jit.load`.
+
+    """
+    model_file_path = os.path.join(bundle_dir, weights_name)
+    if not os.path.exists(model_file_path):
+        if package is None and url is None:
+            raise ValueError(f"To download and load model from source: {source}, `package` must be provided.")
+        if repo is None and url is None:
+            raise ValueError(f"To download and load model from source: {source}, `repo` must be provided.")
+        download(
+            package=package,
+            bundle_dir=bundle_dir,
+            filename=weights_name,
+            source=source,
+            repo=repo,
+            url=url,
+            progress=progress,
+            extract=extract,
+        )
+    # loading with `torch.jit.load`
+    if is_ts_model is True:
+        return torch.jit.load(model_file_path, map_location=map_location)
+    # loading with `torch.load`
+    model_dict = torch.load(model_file_path, map_location=map_location)
+    return model_dict
```

Comment:
I think we can also support to return a instantiated model object with additional `net_name` and `kwargs`, similar to the huggingface torchhub API:
https://github.com/huggingface/transformers/blob/main/hubconf.py#L105

Thanks.